pvlib/modelchain.py

```diff
@@ -961,6 +1035,97 @@ def run_model(self, weather, times=None):
         self.aoi_model()
         self.spectral_model()
         self.effective_irradiance_model()
```

Comment:
Maybe refactor these 3 methods into another method because the same call sequence is used here and below.